### PR TITLE
Introduce CI

### PR DESCRIPTION
**Pull request by hunterjackson**

### CODE REVIEW

File: src/main/java/com/meta/chatbridge/FBID.java [hunk 1]
```diff
@@ -24,7 +24,7 @@ public static FBID from(long value) {
 
   public static FBID from(String value) {
     Objects.requireNonNull(value);
-    long longValue = Long.valueOf(value);
+    long longValue = Long.parseLong(value);
```
Comment (colinmccloskey) [inline src/main/java/com/meta/chatbridge/FBID.java:27]: Don't know the most about Java but this is to return a primitive long value instead of a long object? And valueof has some performance gains from caching?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    - uses: actions/checkout@v3
+    - name: Set up JDK
+      uses: actions/setup-java@v3
+      with:
+        java-version: '20'
+        distribution: 'temurin'
+        cache: maven
+    - name: Build with Maven
+      run: mvn -B package --file pom.xml
+
```
Comment (colinmccloskey) [inline .github/workflows/ci.yml:21]: Maybe add the new messagestack test here?

File: src/test/java/com/meta/chatbridge/store/MessageStackTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    assertThat(ms.messages()).hasSize(1);
+    ms = ms.with(message2);
+    assertThat(ms.messages()).hasSize(2);
+    assertThat(ms.messages()).first().isSameAs(message1);
+    assertThat(ms.messages()).last().isSameAs(message2);
+  }
+
+  @Test
+  void orderCorrection() {
+    Instant start = Instant.now();
```
Comment (colinmccloskey) [inline src/test/java/com/meta/chatbridge/store/MessageStackTest.java:72]: I think we just need the second case in the order correction test, since the first case in both will use the same  messages.stream()
            .sorted(Comparator.comparing(Message::timestamp))
            .collect(Collectors.toUnmodifiableList());
            
But probably still good to have both

File: src/test/java/com/meta/chatbridge/store/MessageStackTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    assertThat(ms.messages()).last().isSameAs(message2);
+
+    ms = MessageStack.of(List.of());
+    ms = ms.with(message2);
+    assertThat(ms.messages()).hasSize(1);
+    ms = ms.with(message1);
+    assertThat(ms.messages()).hasSize(2);
+    assertThat(ms.messages()).first().isSameAs(message1);
+    assertThat(ms.messages()).last().isSameAs(message2);
+  }
```
Comment (colinmccloskey) [inline src/test/java/com/meta/chatbridge/store/MessageStackTest.java:86]: Should we add a test for invalid message? To make sure the assert non-null stays working?

**Review by colinmccloskey — APPROVED** [review]
lgtm

File: src/main/java/com/meta/chatbridge/FBID.java (hunk 1 above)
Comment (hunterjackson) [reply src/main/java/com/meta/chatbridge/FBID.java:27]: Effective Java, Item 61: Prefer primitive types to boxed primitives

File: .github/workflows/ci.yml (hunk 2 above)
Comment (hunterjackson) [reply .github/workflows/ci.yml:21]: this runs all of the tests

File: src/test/java/com/meta/chatbridge/store/MessageStackTest.java (hunk 4 above)
Comment (hunterjackson) [reply src/test/java/com/meta/chatbridge/store/MessageStackTest.java:86]: So when you look at things like this you have to think about what you're testing.  You make a good point that adding a nonnull test here would ensure nobody in the future unwittingly deletes the null check and tries to send null values. However what you actually end up testing is whether null checks work, and if we tested for all of those we'd be adding lots of tests. Ideally for situations like this you do higher level tests that ensure it's impossible to get a null in there at all (which we'll be doing) and the null in the method null check screams if it is possible to get a null in the pipeline.

Comment (hunterjackson) [reply src/test/java/com/meta/chatbridge/store/MessageStackTest.java:86]: We also have type annotations that in theory should warn us in our IDE if it's possible to introduce a null
